api/http/router.go

```diff
@@ -35,38 +34,22 @@ const (
 	PeerIDPath  string = versionedAPIPath + "/peerid"
 )
 
-func setRoutes(h *handler) *handler {
-	h.Mux = chi.NewRouter()
+var router = chi.NewRouter()
 
-	// setup CORS
-	if len(h.options.allowedOrigins) != 0 {
-		h.Use(cors.Handler(cors.Options{
-			AllowedOrigins: h.options.allowedOrigins,
-			AllowedMethods: []string{"GET", "POST", "OPTIONS"},
-			AllowedHeaders: []string{"Content-Type"},
-			MaxAge:         300,
-		}))
-	}
-
-	// setup logger middleware
-	h.Use(loggerMiddleware)
-
-	// define routes
-	h.Get(RootPath, h.handle(rootHandler))
-	h.Get(PingPath, h.handle(pingHandler))
-	h.Get(DumpPath, h.handle(dumpHandler))
-	h.Get(BlocksPath+"/{cid}", h.handle(getBlockHandler))
-	h.Get(GraphQLPath, h.handle(execGQLHandler))
-	h.Post(GraphQLPath, h.handle(execGQLHandler))
-	h.Get(SchemaPath, h.handle(listSchemaHandler))
-	h.Post(SchemaPath, h.handle(loadSchemaHandler))
-	h.Patch(SchemaPath, h.handle(patchSchemaHandler))
-	h.Post(IndexPath, h.handle(createIndexHandler))
-	h.Delete(IndexPath, h.handle(dropIndexHandler))
-	h.Get(IndexPath, h.handle(listIndexHandler))
-	h.Get(PeerIDPath, h.handle(peerIDHandler))
-
-	return h
+func init() {
+	router.Get(RootPath, rootHandler)
+	router.Get(PingPath, pingHandler)
+	router.Get(DumpPath, dumpHandler)
+	router.Get(BlocksPath+"/{cid}", getBlockHandler)
+	router.Get(GraphQLPath, execGQLHandler)
+	router.Post(GraphQLPath, execGQLHandler)
+	router.Get(SchemaPath, listSchemaHandler)
+	router.Post(SchemaPath, loadSchemaHandler)
+	router.Patch(SchemaPath, patchSchemaHandler)
+	router.Post(IndexPath, createIndexHandler)
+	router.Delete(IndexPath, dropIndexHandler)
+	router.Get(IndexPath, listIndexHandler)
+	router.Get(PeerIDPath, peerIDHandler)
```

Comment:
thought: I'm not a big fan of a global var router and setting the routes on package init. I think it would be better for future maintenance (and other beneficial reasons) to keep this under the `setRoutes` function. Although you can simply return the router.

That being said, I like how you moved the handle method to a middleware and that it avoids wrapping the handlers in the `handle` method.

Reply:
`playground.go:init` can also be moved inside the `setRoutes` function.

Reply:
I wanted to keep the `setRoutes` method but I couldn't think of another way to do build time route registration. I'm using the `playground.go:init` function because the playground can be enabled/disabled using the build tag. I'm happy to move everything back if we can think of an alternative that works with build tags.

Reply:
Thanks! My brain was stuck on that for a while. I think I came up with a good solution. The route registration is moved back to `setRoutes` and the playground handler is set at build time via `init`. I made sure it works when not including the playground as well.